pymongo/asynchronous/mongo_client.py

```diff
@@ -1115,16 +1224,30 @@ def options(self) -> ClientOptions:
         """
         return self._options
 
+    def eq_props(self) -> tuple[tuple[_Address, ...], Optional[str], Optional[str], str]:
+        return (
+            tuple(sorted(self._resolve_srv_info["seeds"])),
+            self._options.replica_set_name,
+            self._resolve_srv_info["fqdn"],
+            self._resolve_srv_info["srv_service_name"],
+        )
+
     def __eq__(self, other: Any) -> bool:
         if isinstance(other, self.__class__):
-            return self._topology == other._topology
+            if self._topology is None:
+                return self.eq_props() == other.eq_props()
+            else:
+                return self._topology == other._topology
         return NotImplemented
 
     def __ne__(self, other: Any) -> bool:
         return not self == other
 
     def __hash__(self) -> int:
-        return hash(self._topology)
+        if self._topology is None:
```

Comment:
We still need to address hashing and equality. It's broken to change the hash of an object since it means that a single client could be added twice to a dictionary or set. The hash (and eq) methods need to use a single deterministic approach:
```python
set = {}
c = AsyncMongoClient('mongodb+srv://...')
set.add(c)
await c.aconnect()
set.add(c)
assert len(set) == 1
```

This means we need to use the eq_props() approach in all cases.